src/iqm_client/iqm_client.py

```diff
@@ -550,27 +575,38 @@ def submit_circuits(
         Returns:
             ID for the created task. This ID is needed to query the status and the execution results.
         """
-        serialized_qubit_mapping: Optional[list[SingleQubitMapping]] = None
-        if qubit_mapping is not None:
-            # check if qubit mapping is injective
-            target_qubits = set(qubit_mapping.values())
-            if not len(target_qubits) == len(qubit_mapping):
-                raise ValueError('Multiple logical qubits map to the same physical qubit.')
-
-            # check if qubit mapping covers all qubits in the circuits
-            for circuit in circuits:
+        serialized_qubit_mappings: Optional[BatchQubitMapping] = None
+        if qubit_mappings is not None:
+            # check that the number of qubit mappings matches the number of circuits
+            if len(qubit_mappings) != len(circuits):
+                raise ValueError('The number of qubit mappings does not match the number of circuits.')
+
+            # check if all qubit mappings map to same physical qubits
+            if len(set(frozenset(m.values()) for m in qubit_mappings)) != 1:
+                raise ValueError('All qubit mappings should map to the same set of physical qubits.')
+
+            # check if all qubit mappings are injective
+            for qubit_mapping in qubit_mappings:
+                target_qubits = set(qubit_mapping.values())
+                if not len(target_qubits) == len(qubit_mapping):
+                    raise ValueError('Multiple logical qubits map to the same physical qubit.')
+
+            # check if each qubit mapping covers all qubits in the corresponding circuit
+            for i, (qubit_mapping, circuit) in enumerate(zip(qubit_mappings, circuits)):
                 circuit_qubits = circuit.all_qubits()
                 diff = circuit_qubits - set(qubit_mapping.keys())
```

Comment:
```suggestion
                diff = circuit_qubits - set(qubit_mapping)
```
By default dicts are iterated over the keys.